modules/Queue.js

```diff
@@ -39,15 +39,16 @@ class Queue extends Model
             throw 'A task and task type is required!';
         
         let oldTail = await Queue.query().where('group_type', '=', task.group_type).andWhere('is_active', '=', 1).whereNull('next').first();
-        
-        let newTail;
+        let newTail, is_head = oldTail ? 0 : 1;
         if (task.uuid)
```

Comment:
This check allows me to recycle task records if I want to. Instead of creating a whole new entry.